pkg/cli/destroy/destroy.go

```diff
@@ -319,19 +321,45 @@ func deleteAllResourcesAndWait(ctxWithoutDeadline, ctxWithDeadLine context.Conte
 						return false, fmt.Errorf("failed to unmarshal virtual cluster config for %v %q: %w", resource, namespacedName, err)
 					}
 					if vConfig.ControlPlane.BackingStore.Database.External.Connector != "" {
-						log.Warnf("IMPORTANT! You are removing an externally deployed virtual cluster %q from the platform.\n It will not be destroyed as the deployment is managed externally, but its database will be removed rendering it inoperable.", namespacedName)
-						if !nonInteractive {
-							yesOpt := "yes"
-							noOpt := "no"
-							out, err := log.Question(&survey.QuestionOptions{
-								Options:  []string{yesOpt, noOpt},
-								Question: "Do you want to continue?",
-							})
-							if err != nil {
-								return false, fmt.Errorf("failed to prompt for confirmation: %w", err)
+						// check if already marked for deletion to avoid getting repeated prompts in case of backoff-retry scenario
+						if object.GetDeletionTimestamp().IsZero() {
+							log.Warnf("IMPORTANT! You are removing an externally deployed virtual cluster %q from the platform.\n It will not be destroyed as the deployment is managed externally, but its database will be removed rendering it inoperable.", namespacedName)
+							if !nonInteractive {
+								resp, err := log.Question(&survey.QuestionOptions{
+									Options:  []string{util.PositiveResponse, util.NegativeResponse},
+									Question: "Do you want to continue?",
+								})
+								if err != nil {
+									return false, fmt.Errorf("failed to prompt for confirmation: %w", err)
+								}
+								if resp != util.PositiveResponse {
+									return false, fmt.Errorf("destroy cancelled during prompt")
+								}
 							}
-							if out != yesOpt {
-								return false, fmt.Errorf("destroy cancelled during prompt")
+						}
+
+						// if db-connector secret is not found then the controller won't be able to remove the "cleanup-connectors" finalizer
+						// and the destroy op gets stuck. Hence, the user needs to opt-in for the removal of finalizer
+						for _, cond := range virtualClusterInstance.Status.Conditions {
+							if cond.Type == util.DBConnectorSecretNotFound && cond.Status == corev1.ConditionTrue {
```

Comment:
As mentioned in [this PR](https://github.com/loft-sh/loft-enterprise/pull/4116), I'd like to see a more generic `DbConnectorSynced` condition type, and use a reason to communicate the more specific failure to check here.